pkg/gce-pd-csi-driver/controller.go

```diff
@@ -138,6 +143,11 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("CreateVolume replication type '%s' is not supported", replicationType))
 	}
 
+	if _, alreadyExists := gceCS.volumes.LoadOrStore(volKey.String(), true); alreadyExists {
```

Comment:
It might be nice to wrap this into a new Interface instead of operating directly on sync.Map.  That way in the future, we can enhance this (to add things like timing/metrics measurements and long operation detection).